src/librustc/ty/context.rs

```diff
@@ -482,6 +482,23 @@ impl<'tcx> TypeckTables<'tcx> {
         }
     }
 
+    pub fn opt_type_dependent_def(&self, id: HirId) -> Option<&Def> {
+        validate_hir_id_for_typeck_tables(self.local_id_root, id, false);
+        self.type_dependent_defs.get(&id.local_id)
+    }
+
+    pub fn opt_type_dependent_def_id(&self, id: HirId) -> Option<DefId> {
+        match self.opt_type_dependent_def(id) {
+            Some(def) => Some(def.def_id()),
+            _ => None,
+        }
+    }
+
+    pub fn type_dependent_def_id(&self, id: HirId) -> DefId {
```

Comment:
since this seems unused, you can remove it and have clippy just unwrap the `opt_type_dependent_def_id` (and maybe rename that method to drop the `opt_`)